schema.go

```diff
@@ -211,28 +210,25 @@ func (n name) FullName() string {
 }
 
 type fingerprinter struct {
-	fingerprint [32]byte
-	cache       map[FingerprintType][]byte
+	fingerprint atomic.Value
+	cache       sync.Map
```

Comment:
```suggestion
	fingerprint atomic.Value   // [32]byte
	cache       concurrent.Map //map[FingerprintType][]byte
```

`concurrent.Map` should be used to allow for use pre go1.9, it is also nice to have the types in comments next to the abstractions.

Reply:
I think you will need to remove the `sync` import as well for the above suggestion.

Reply:
I fixed `sync.Map` and added the comments.
https://github.com/hamba/avro/pull/76/commits/026dae629edddd3c2280dcaeed1be8f04ecd5405
Benchmark results look the same as before.
```
goos: darwin
goarch: amd64
pkg: github.com/nrwiersma/avro-benchmarks
BenchmarkGoAvroDecode-4     	  335384	      3156 ns/op	     442 B/op	      27 allocs/op
BenchmarkGoAvroEncode-4     	  308666	      4386 ns/op	     834 B/op	      63 allocs/op
BenchmarkHambaDecode-4      	 2443789	       544 ns/op	      64 B/op	       4 allocs/op
BenchmarkHambaEncode-4      	 3048300	       391 ns/op	     112 B/op	       1 allocs/op
BenchmarkLinkedinDecode-4   	  676179	      1775 ns/op	    1728 B/op	      35 allocs/op
BenchmarkLinkedinEncode-4   	 1829768	       659 ns/op	     248 B/op	       5 allocs/op
PASS
ok  	github.com/nrwiersma/avro-benchmarks	10.144s
```